src/accountsdb/db.zig

```diff
@@ -703,11 +704,12 @@ pub const AccountsDB = struct {
     /// combines multiple thread indexes into the given index.
     /// each bin is also sorted by pubkey.
     pub fn combineThreadIndexesMultiThread(
-        logger: Logger,
+        logger_: Logger,
         index: *AccountIndex,
         thread_dbs: []const AccountsDB,
         task: sig.utils.thread.TaskParams,
     ) !void {
+        const logger = logger_.withScope(@src().fn_name);
```

Comment:
imo i could see the pros of knowing what method is currently being executed which you can ctrl+f find in the codebase and know exactly where its executing -- if its not too much work the ideal would be something like `AccountsDB.combineThreadIndexesMultiThread`

Reply:
Where do you draw the line? Would you rescope the logger for every function?

The information comes with a tradeoff of visual noise and code complexity. The costs are not too high, but neither is the benefit since you can also search for the log message itself.

In terms of understanding the logs on their own, I feel the file/struct/component name is much more useful than the function name, but I could see the function name providing valuable depth in some situations.

Reply:
There could be a concept of nesting scope. for example:
```zig
const global_logger = logger.withScope("global");
const local_logger = global_logger.addScope("local");
```
Using local_logger would result in a scope like "global.local".

If you want to rescope the logger at a very granular level, you could add some functions like `infoScoped` that take a scope parameter.

Reply:
> There could be a concept of nesting scope. 

Indeed. This can be a follow up task, if it is agreed as a new feature to be added to the logger.

Reply:
My opinion is that we should never do this:
```zig
const logger = logger_.withScope(@src().fn_name);
```
...because it *only* includes the name of the function, which is probably not enough information for most people to understand where it actually comes from. The logs will be confusing with just a bunch of individual function names. I think it's more valuable to have higher level scopes like "gossip" and "accountsdb" so you can take a quick glance at the logs to figure out what's going on. If you want to know exactly where a log is coming from you can always search the code for the log message.

I'm fine with fully-qualified scoping, where the function name comes after the file name or struct name, but I think it's too obscure to have *only* the function name.

If you're going to do fully qualified scope, personally I think nesting scope would be the easiest way to do it. I don't think this needs to be a follow up task. It only adds a few lines of code.
```zig
pub fn addScope(self: Self, comptime new_scope: []const u8) ScopedLogger(scope ++ new_scope) {
    return self.withScope(scope ++ new_scope);
}
```

Reply:
So I don't think the nesting scope feature solves the problem that led to the use of `@src().fn_name` in the first place. 

The `@src().fn_name` was introduced as the scope to to use for free standing functions that are not part of a struct. 

For logging that occurs within a struct, the scope for that is quite clear, it is scoped to `@typeName(Self)`, but for free standing function, it is not that clear, hence the idea to scope it to `@src().fn_name`.

But given the cons of that approach, I have removed the usage of `@src().fn_name` by using a scope constant defined in the file. Ref: [45affc0](https://github.com/Syndica/sig/pull/345/commits/45affc0daea265c58a86c773f8320b7402819004)

Another approach might be to have the logger typed as `anytype` and leave it to the caller to provide either a scoped logger or un-scoped logger. But went with the using a scope constant instead. 

The nested scope might still be a feature to consider (although might have to be evaluated against the current setup where upscoped logger is always used in the boundary where a logger is passed in, into a struct - where it then get scoped appropriately) .

Reply:
Sounds good to me. Feel free to resolve this conversation. I'm just leaving it open for visibility.

> But given the cons of that approach, I have removed the usage of @src().fn_name by using a scope constant defined in the file. Ref: [45affc0](https://github.com/Syndica/sig/pull/345/commits/45affc0daea265c58a86c773f8320b7402819004)

I like the idea of scoping by file. To me that feels like the most "normal" way of doing scoping. Initially, I assumed all loggers would be scoped by file. But scoping by struct also seems good.

I'm not saying we need to add the nested scope now. I just misunderstand why you used function names as the scope. I assumed that it was because you wanted extra granularity. I was just suggesting the nested scope as a way to include the granularity without sacrificing high level context.

> The nested scope might still be a feature to consider (although might have to be evaluated against the current setup where upscoped logger is always used in the boundary where a logger is passed in, into a struct - where it then get scoped appropriately) .

Let's say you have a file with one public function and five private functions. Since the five are private, they are all in some way called by the public function. Let's assume all of those private functions need a logger. You can define a single logger scope for the entire file. I would expect the public function to accept an unscoped logger as a parameter, and then it would convert it to a scoped logger. Then all the private functions would accept the *scoped* logger as a parameter. That way you don't need to pointlessly convert back and forth between the scoped and unscoped variants of the logger.

Now, let's say you wanted to increase the granularity of the scope in the logs for each individual logger by including the function name in the scope. I think that would be best served by appending the scopes together, so you don't lose the filename from the scope. This is where the idea of "nesting scope" could come into play.

Reply:
> Let's say you have a file with one public function and five private functions. Since the five are private, they are all in some way called by the public function. Let's assume all of those private functions need a logger. You can define a single logger scope for the entire file. I would expect the public function to accept an unscoped logger as a parameter, and then it would convert it to a scoped logger. Then all the private functions would accept the scoped logger as a parameter. That way you don't need to pointlessly convert back and forth between the scoped and unscoped variants of the logger

That is indeed a valid scenario having the nested scoping helps. I'll add a task for this so it does not fall through the cracks.